paper/paper.md

```diff
@@ -0,0 +1,126 @@
+
+![Converter workflow for nyaml tool, XML to YAML and vice-versa \label{fig:nyaml_workflow}](assets/workflow-1.pdf){ width=75% }
+
+Conversion from __YAML to XML__ follows specific workflow steps (depicted in \autoref{fig:nyaml_workflow}) according to NXDL rules and syntax specific to the YAML format (see [documentation](https://fairmat-nfdi.github.io/nyaml)). Starting from a given input YAML file (see \autoref{fig:nyaml_workflow}), the workflow performs the following steps:
+
+1. The __nyaml2nxdl__ converter collects the input __.yaml__ file.
+
+2. Using __PyYAML__ [@PyYAML:2024], the converter collects and tracks comments in the YAML file.
+
+3. The converter parses the YAML file into a nested hashed map—a Python dictionary object.
+
+4. The converter writes the hashed map and comments into an output XML file in accordance with NXDL concepts.
+
+
+The conversion algorithm interprets the specific keywords and syntactic rules to transcode the NXDL from YAML into XML (see [documentation](https://fairmat-nfdi.github.io/nyaml)). Leveraging the NXDL rules, the conversion process detects possible inconsistencies in the YAML content and raises errors or warnings if the rules are not properly followed.
+
+
+The __XML to YAML__ conversion also follows a well-defined data workflow (\autoref{fig:nyaml_workflow}) that converts a given input XML file into a YAML file. The workflow begins with the XML input and proceeds as follows:
+
+
+1. The __nxdl2nyaml__ converter takes over the __.nxdl.xml__ file.
+
+2. Using __lxml__ [@lxml:2025], the converter parses the XML file and builds an XML tree structure.
+
+3. Applying YAML-specific keywords and formatting rules, the converter generates a YAML file from the XML tree.
+
+4. The converter computes a SHA256 hash of the generated YAML content.
+
+5. The converter writes the YAML file and appends both the hash and the original XML content as comments at the end of the YAML file.
+
+
+By attaching the hash and the original XML content to the YAML output (__.yaml__ file), the tool enables lossless round-trip conversions — provided the YAML content remains unchanged. That is, if the YAML is not modified, converting back from YAML to XML, the original commented XML content will be written back to the XML file without any modification. However, if the YAML content is modified, the XML tree will be reconstructed from the YAML content and written to the XML file, which will differ from the original XML content included in the comments. This caching approach streamlines the XML → YAML → XML workflow and facilitates straightforward comparisons of XML files in version control systems such as Git.
```

Comment:
see comment I made on the docstring branch https://github.com/FAIRmat-NFDI/nyaml/pull/33/files#r2202707647